test/unit/sql/data-types.test.ts

```diff
@@ -0,0 +1,1492 @@
+          dataType: DataTypes.MEDIUMINT(2).UNSIGNED.ZEROFILL,
+          expect: {
+            default: zeroFillUnsupportedError,
+            'mariadb mysql': 'MEDIUMINT(2) UNSIGNED ZEROFILL',
+          },
+        },
+        {
+          title: 'MEDIUMINT.ZEROFILL',
+          dataType: DataTypes.MEDIUMINT.ZEROFILL,
+          expect: {
+            default: zeroFillUnsupportedError,
+            'mariadb mysql': 'MEDIUMINT ZEROFILL',
+          },
+        },
+        {
+          title: 'MEDIUMINT(2).ZEROFILL',
+          dataType: DataTypes.MEDIUMINT(2).ZEROFILL,
+          expect: {
+            default: zeroFillUnsupportedError,
+            'mariadb mysql': 'MEDIUMINT(2) ZEROFILL',
+          },
+        },
+        {
+          title: 'MEDIUMINT.ZEROFILL.UNSIGNED',
+          dataType: DataTypes.MEDIUMINT.ZEROFILL.UNSIGNED,
+          expect: {
+            default: zeroFillUnsupportedError,
+            'mariadb mysql': 'MEDIUMINT UNSIGNED ZEROFILL',
+          },
+        },
+        {
+          title: 'MEDIUMINT(2).ZEROFILL.UNSIGNED',
+          dataType: DataTypes.MEDIUMINT(2).ZEROFILL.UNSIGNED,
+          expect: {
+            default: zeroFillUnsupportedError,
+            'mariadb mysql': 'MEDIUMINT(2) UNSIGNED ZEROFILL',
+          },
+        },
+      ];
+
+      for (const row of cases) {
+        testsql(row.title, row.dataType, row.expect);
+      }
+
+      describe('validate', () => {
+        it('should throw an error if `value` is invalid', () => {
+          const type = DataTypes.MEDIUMINT();
+
+          expect(() => {
+            type.validate('foobar');
+          }).to.throw(ValidationErrorItem, `'foobar' is not a valid mediumint`);
+
+          expect(() => {
+            type.validate(123.45);
+          }).to.throw(ValidationErrorItem, '123.45 is not a valid mediumint');
+        });
+
+        it('should not throw if `value` is an integer', () => {
+          const type = DataTypes.MEDIUMINT();
+
+          expect(() => type.validate(-8_388_608)).not.to.throw();
+          expect(() => type.validate('8388607')).not.to.throw();
+        });
+      });
+    });
+
+    describe('BIGINT', () => {
+      const zeroFillUnsupportedError = new Error(`${dialectName} does not support the BIGINT.ZEROFILL data type.
+See https://sequelize.org/docs/v7/other-topics/other-data-types/ for a list of supported data types.`);
+      const unsignedUnsupportedError = new Error(`${dialectName} does not support the BIGINT.UNSIGNED data type.
+See https://sequelize.org/docs/v7/other-topics/other-data-types/ for a list of supported data types.`);
+
+      testsql('BIGINT', DataTypes.BIGINT, {
+        default: 'BIGINT',
+      });
+
+      testsql('BIGINT.UNSIGNED', DataTypes.BIGINT.UNSIGNED, {
+        default: unsignedUnsupportedError,
+        'mysql mariadb': 'BIGINT UNSIGNED',
+      });
+
+      testsql('BIGINT.UNSIGNED.ZEROFILL', DataTypes.BIGINT.UNSIGNED.ZEROFILL, {
+        default: unsignedUnsupportedError,
```

Comment:
Do we want to have a specific error for when both `UNSIGNED` and `ZEROFILL` are not supported?

Reply:
I'd prefer not to by laziness but will if requested

Reply:
Then we'll just leave it like this for now.